File: java/udf/WordCountMapper.java
package udf;

import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

import mr.Mapper;
import mr.Pair;

public class WordCountMapper implements Mapper<String, Integer> {

    @Override
    public List<Pair<String, Integer>> map(String line) {
        String[] words = line.split(" ");
        return Arrays.stream(words).map(word -> new Pair<>(word, 1)).collect(Collectors.toList());
    }
}

File: golang/udf/wordcount_mapper.go
package main

import (
	"mr/mr"
	"strings"
)

var Map mr.Map = func(line string) []*mr.Pair {
	words := strings.Split(line, " ")
	pairs := make([]*mr.Pair, 0, len(words))
	for _, word := range words {
		pairs = append(pairs, &mr.Pair{Key: word, Value: 1})
	}
	return pairs
}
